autometa/binning/recursive_dbscan.py

```diff
@@ -522,23 +523,26 @@ def binning(
         i.e. [taxid,superkingdom,phylum,class,order,family,genus,species]
     markers : pd.DataFrame
         wide format, i.e. index=contig cols=[marker,marker,...]
-    domain : str
+    domain : str, optional
         Kingdom to determine metrics (the default is 'bacteria').
         choices=['bacteria','archaea']
-    completeness : float
+    completeness : float, optional
         Description of parameter `completeness` (the default is 20.).
-    purity : float
+    purity : float, optional
         Description of parameter `purity` (the default is 90.).
-    taxonomy : bool
+    taxonomy : bool, optional
         Split canonical ranks and subset based on rank then attempt to find clusters (the default is True).
         taxonomic_levels = [superkingdom,phylum,class,order,family,genus,species]
-    method : str
+    starting_rank : str, optional
+        Starting canonical rank at which to begin subsetting taxonomy (the default is superkingdom).
+        Choices are superkingdom, phylum, class, order, family, genus, species.
+    method : str, optional
         Clustering `method` (the default is 'dbscan').
         choices = ['dbscan','hdbscan']
-    reverse : bool
-        True - [superkingdom,phylum,class,order,family,genus,species]
-        False - [species,genus,family,order,class,phylum,superkingdom]
-    verbose : bool
+    reverse_ranks : bool, optional
+        False - [superkingdom,phylum,class,order,family,genus,species] (Default)
+        True - [species,genus,family,order,class,phylum,superkingdom]
```

Comment:
Just checking - did you intent to make `reverse_ranks` have opposite meanings for `True` and `False` compared to `reverse`?

Reply:
It does seem to be internally consistent, but it is a little weird because later `NCBI.CANONICAL_RANKS` is reversed if `reverse_ranks == False`. This will probably trip contributors up.

Reply:
Yes, I reversed the logic thinking this would be more intuitive to the end user. `NCBI.CANONICAL_RANKS` is by default ordered from most to least specific but as described in the NAR paper, the order splits by taxonomic rank from superkingdom down to species. The other way around this is to change the `NCBI.CANONICAL_RANKS` order and reverse the order in other files where it is used. I've placed comments at these lines of code to clarify to any developer.